### ISSUE TITLE
Whitened raw space for parameters under a joint prior

### ISSUE THREAD

**Issue by gvcallen** [gvcallen/paramrf#194]
## What to build

A whitened raw space for parameters under a joint prior, following ADR-0005 (`docs/adr/0005-priors-by-name.md`). Optimisers and samplers move in the distribution's well-conditioned latent space, e.g. a normalizing flow's base `z`, instead of in the space the distribution was stated in.

- The whitening is inferred the same way as for a one-dimensional prior, with `parax.constraints.infer_distribution_constraint`: a flow's base and bijector, the Cholesky factor of a multivariate normal, and so on. When no whitening is known, raw stays the distribution's own space.
- `prf.param_values(..., space='raw')` and `prf.update(..., space='raw')` go through the whitening for these parameters. Each parameter's raw value is its entry in the whitened vector, so names are unchanged.
- `log_prior(space='raw')` carries the matching change-of-variables term.
- Parameters not under a joint prior are unaffected.

### Bounds

Whitened raw coordinates reach every point of the distribution's support. A parameter still cannot hold a value outside its bounds, so the support must already lie inside them:

- **`space='raw'`**: nothing to check. The whitening ends in the parameters' old raw space, which maps into their bounds through their old raw-to-declared mappings. This is the intended pipeline: a `fleqx` flow trained on an earlier fit's raw samples. Its support is all of Rⁿ, and it gives an exactly normalised prior, so the evidence stays exact.
- **`space='declared'` or `'physical'`**: `prf.prior` raises at attach unless the distribution's support lies inside the selected parameters' current bounds, taken to the prior's space. The error names the parameters and points to `space='raw'`. Such a distribution cannot be truncated and renormalised exactly, and an unnormalised one would bias the evidence.

The check uses the current bounds, which do not record whether they come from a model field (validity) or from the user (a range). Relaxing it to validity only is follow-up work.

Seam: `pmrf/parameters.py`, `pmrf/_solver_view.py`.

## Acceptance criteria

- [ ] For the correlated-Gaussian example (`L z + mu`), the raw values of `a.R` and `b.R` equal `L⁻¹(x - mu)`. Here `x` is their values in the joint prior's stated space.
- [ ] `prf.update(m, prf.param_values(m, space='raw'), space='raw')` returns `m` for a model with a joint prior.
- [ ] `log_prior(space='raw')` is consistent with `log_prior(space='declared')` through the Jacobian. This is checked with a finite-difference or `jax.jacobian` test.
- [ ] MCMC and a MAP fit on the correlated-Gaussian example give the same answers as before this ticket, within tolerance.
- [ ] A distribution with no known whitening still works, with raw equal to its own space.
- [ ] A declared-space joint prior whose support leaves the parameters' bounds (e.g. the correlated Gaussian stated in declared space over bounded parameters) raises at attach, naming the parameters and `space='raw'`. A declared-space one whose support lies inside the bounds is accepted.
- [ ] No solver step can write a value outside a parameter's bounds, so no test needs an `xfail` for it. `test_minimizer_moves_a_parameter_under_a_joint_prior` moves to `space='raw'`, and the attach error has its own test.

## The correlated-Gaussian example

An affine bijector stands in for a trained flow. It has the same structure (a bijector over a standard-normal base) and a closed-form answer. From the handoff that motivated ADR-0005:

```python
model = {
    'a': Resistor(R=prf.Random(RTNormal(50.0, 0.1)), name='a'),
    'b': Resistor(R=prf.Random(RTNormal(50.0, 0.1)), name='b'),
    'c': Capacitor(C=prf.Random(RTNormal(1.0, 0.1), scale=1e-12), name='c'),
}
names = ('a.R', 'b.R')                       # c.C keeps its own prior
mu = jnp.array([3.9, 3.9])
L = jnp.array([[0.10, 0.0], [0.08, 0.06]])   # correlation 0.8
dist = Transformed(Independent(Normal(jnp.zeros(2), jnp.ones(2))), Chain([Block(Shift(mu), 1), TriangularLinear(L)]))
model = prf.prior(model, names, dist, space='raw')
```

## Blocked by

- #193
